### ISSUE TITLE
Improve accessibility to the authors select box

### ISSUE THREAD

**Issue by andergmartins** [publishpress/PublishPress-Authors#63]
User's feedback:

>The Author management interface on a single post is *super*
inaccessible. There are more issues than I can list, but the big issues are
that it's not keyboard accessible and doesn't have accessible labels. The
"remove author" "x" should be a button element with screen reader text, for
example. Select2, while undoubtedly very nice for us sighted mouse users,
is also quite inaccessible.

**Comment by ojopaul** [comment]
@stevejburge This issue is same as https://github.com/publishpress/PublishPress-Authors/issues/89 and I'm not sure is relevant again according to https://github.com/publishpress/PublishPress-Authors/issues/89#issuecomment-956574559 or I probably missed the issue

**Comment by stevejburge** [comment]
@ojopaul OK, thanks
